platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/PlatformBuilder.java

```diff
@@ -0,0 +1,375 @@
+     * @return this
+     */
+    @NonNull
+    public PlatformBuilder withConfigurationBuilder(@Nullable final ConfigurationBuilder configurationBuilder) {
+        this.configurationBuilder = configurationBuilder;
+        return this;
+    }
+
+    /**
+     * The path to the settings file. Default is "./settings.txt" where "./" is the current working directory of the
+     * JVM.
+     *
+     * @param path the path to settings.txt
+     * @return this
+     * @throws IllegalArgumentException if the path does not exist
+     */
+    public PlatformBuilder withSettingsPath(@NonNull final Path path) {
+        Objects.requireNonNull(path);
+        final Path absolutePath = getAbsolutePath(path);
+        if (!Files.exists(absolutePath)) {
+            throw new IllegalArgumentException("File " + absolutePath + " does not exist");
+        }
+        this.settingsPath = absolutePath;
+        return this;
+    }
+
+    /**
+     * The path to the config file. Default is "./config.txt" where "./" is the current working directory of the JVM.
+     *
+     * @param path the path to config.txt
+     * @return this
+     * @throws IllegalArgumentException if the path does not exist
+     */
+    public PlatformBuilder withConfigPath(@NonNull final Path path) {
+        Objects.requireNonNull(path);
+        final Path absolutePath = getAbsolutePath(path);
+        if (!Files.exists(absolutePath)) {
+            throw new IllegalArgumentException("File " + absolutePath + " does not exist");
+        }
+        this.configPath = absolutePath;
+        return this;
+    }
+
+    /**
+     * Build a platform but do not start it.
+     *
+     * @return a new platform instance
+     */
+    public SwirldsPlatform build() {
+        return build(false);
+    }
+
+    /**
+     * Setup static utilities. If running multiple platforms in the same JVM, this should be called when constructing
+     * the first platform only.
+     *
+     * @param configuration the configuration for this node
+     * @return true if this is the first time this method has been called, false otherwise
+     */
+    private boolean setupStaticUtilities(@NonNull final Configuration configuration) {
+        if (staticSetupCompleted) {
+            // Only setup static utilities once
+            return false;
+        }
+        staticSetupCompleted = true;
+
+        ConfigurationHolder.getInstance().setConfiguration(configuration);
+
+        // Setup logging, using the configuration to tell us the location of the log4j2.xml
+        final PathsConfig pathsConfig = configuration.getConfigData(PathsConfig.class);
+        final Path log4jPath = pathsConfig.getLogPath();
+        try {
+            Log4jSetup.startLoggingFramework(log4jPath).await();
+        } catch (final InterruptedException e) {
+            CommonUtils.tellUserConsole("Interrupted while waiting for log4j to initialize");
+            Thread.currentThread().interrupt();
```

Comment:
We should throw an exception in that point from my point of view

Reply:
done